main.sh

```diff
@@ -16,18 +16,38 @@ SERVER_URL="${SERVER_URL%/}"
 
 RETRY_COUNT=${RETRY_COUNT:-10}
 
-echo "Attempting to get status for commit $COMMIT_SHA from $SERVER_URL with $RETRY_COUNT retries"
-Response=$(
-  curl \
+if [ -z "$PROJECT_ID" ]; then
+  echo "::warning::'project_id' is not provided. This will become required for newer projects. For legacy projects this will continue to work as is. It is recommended to supply your 'project_id'".
```

Comment:
```suggestion
  echo "::warning::'project_id' was not provided. 'project_id' will become required for newer projects. For legacy projects this will continue to work as is. It is recommended to supply your 'project_id'".
```